### DISCUSSION TITLE
Xdrip frequently "losing connection" on xiaomi Redmi Note 8 Pro

### DISCUSSION THREAD

**Discussion by jengleruk** [NightscoutFoundation/xDrip#3571] [Bug reports]
Just installed xdrip on xiaomi Redmi Note 8 Pro for the first time (previously was using the patched Dexcom app).

Have followed all the settings recommendations, but we are finding that the app appears to go to sleep regularly.  Even when literally right next to the sensor, we can see from AAPS that it stops receiving readings.

When we go to Xdrip, it reports readings missed, and that it has been "aggressively restarting collector due to lack of reception" every 5 or 6 minutes.

BUT, each time we go to the xDrip home page, reception is restored for the next reading (and missed readings are backfilled - all sent to AAPS in one go).  So it seems to be the act of viewing the app itself which cures the lack of reception.

Does this story give any clues as to what settings I might try looking at?

Thank you

![WhatsApp Image 2024-07-15 at 19 29 32_08befc27](https://github.com/user-attachments/assets/a497ddd8-ed16-484d-9d31-85552a8760df)

**Comment by Navid200** [comment]
Please post a screenshot of the following two pages: 

Dex Status page by just swiping left on the status page you have shown 

and 

Settings > Dex Debug Settings

**Reply by jengleruk** [reply]
DEx debug settings sent as 2 pics as needed to scroll down to see bottom - doesn't all fit on one page

![WhatsApp Image 2024-07-15 at 20 38 40_bd06e522](https://github.com/user-attachments/assets/59120136-3f1e-4bd6-9924-98ec9ddf5bbc)
![WhatsApp Image 2024-07-15 at 20 38 41_5d2cc309](https://github.com/user-attachments/assets/2c2c50cb-b2d0-433c-b1db-d4c6ae02e6fe)
![WhatsApp Image 2024-07-15 at 20 38 41_ac0cb040](https://github.com/user-attachments/assets/a0f496ad-753b-4d37-95df-eb68248a2519)

**Reply by jengleruk** [reply]
BTW had resolved when I sent that - it isn't in the "not working" state as at that photo

**Reply by Navid200** [reply]
Can you show the logs?

**Reply by jengleruk** [reply]
please can you advise which logs - how do I do that?  Thanks

**Reply by Navid200** [reply]
Sure: https://navid200.github.io/xDrip/docs/Logs.html

**Reply by jengleruk** [reply]
sent - thank you!

**Reply by Navid200** [reply]
Would you please just take a screenshot and show the latest logs by posting here?  
May be you uploaded the logs.  I won't get them that way, sorry.

**Reply by jengleruk** [reply]
Ah, OK. Here you go....
![WhatsApp Image 2024-07-15 at 22 59 03_b9212f12](https://github.com/user-attachments/assets/3ce71606-40de-4ce9-b884-9e7748c7e04a)

**Reply by Navid200** [reply]
Please have a look at this:  https://dontkillmyapp.com/xiaomi

Is there anything on that page that can help you disable any feature on your phone that could disable xDrip running in the background?
xDrip needs to run in the background.  If the phone interferes with that, xDrip will not be able to wake up when it needs to in order to get a new reading once every 5 minutes.

**Reply by jengleruk** [reply]
Thanks. Yes, I found that background autostart was not enabled.  Also, MIUI optimisations were enabled, so swiched that off.  I will monitor carefully now and see what happens!  Thank you again.

**Reply by jengleruk** [reply]
Thanks for all your help. Those interventions appear to have cured the issue.

**Reply by Navid200** [reply]
You're welcome.
Thanks for the feedback
